### ISSUE TITLE
Make "time to wait after a new file version before computing diagnostics" configurable

### ISSUE THREAD

**Issue by AleksanderPasiut** [clangd/clangd#1471]
When analyzed file is being edited, clangd starts parsing it almost right away, with minimum timeout. As a result it consumes lots of cpu and shows lots of errors for nothing, before the edition is finished. It would be nice if that timeout could be configurable, so that clangd waits a moment before parsing the file that changed.

As far as I understand, main effort would be to make variables in this file: https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clangd/ClangdServer.h in lines 149 and 150 configurable, instead of hardcoded.

**Comment by HighCommander4** [comment]
See also https://github.com/clangd/vscode-clangd/issues/262 which links to patches that take a different approach to address this on the client side in VSCode.

**Comment by strpbrk** [comment]
I believe this is a client (VSCode) problem. It should not send changes to the server with every keystroke. Even if patched in the server, I still don't want each keystroke to be sent to the server.

**Comment by Iritscen** [comment]
I agree that the best place to patch this is at the source, VS Code.  People occasionally file tickets on Code asking for a delay feature to be added which will apply to all Intellisense, linters, LSP, etc., and they haven't had any luck, but there is a new ticket as of three days ago, so I am piggybacking on that one hoping for a different outcome this time.  I would love for this ticket to remain open though so we can consider all our options for solving what I think is a pretty vexing behavior issue.

**Comment by HighCommander4** [comment]
> there is a new ticket as of three days ago, so I am piggybacking on that one hoping that something different happens this time

Could you post a link please for others who might be interested in following that ticket?

**Comment by Iritscen** [comment]
Yes, it's at [https://github.com/microsoft/vscode/issues/251470](https://github.com/microsoft/vscode/issues/251470).

**Comment by Iritscen** [comment]
No movement on that VS Code ticket, but I thought I would mention that I implemented a proposed fix from the side of the clangd Code plugin here: https://github.com/clangd/vscode-clangd/issues/262#issuecomment-3067711697
